src/CodeGen_LLVM.cpp

```diff
@@ -1192,6 +1192,107 @@ void CodeGen_LLVM::optimize_module() {
         module->print(dbgs(), nullptr, false, true);
     }
 
+    std::unique_ptr<TargetMachine> tm = make_target_machine(*module);
+
+#if LLVM_VERSION >= 90
+    PipelineTuningOptions pto;
+    pto.LoopInterleaving = !get_target().has_feature(Target::DisableLLVMLoopUnroll);
+    pto.LoopVectorization = !get_target().has_feature(Target::DisableLLVMLoopVectorize);
+    pto.LoopUnroll = pto.LoopInterleaving;
+    // Clear ScEv info for all loops. Certain Halide applications spend a very
+    // long time compiling in forgetLoop, and prefer to forget everything
+    // and rebuild SCEV (aka "Scalar Evolution") from scratch.
+    // Sample difference in compile time reduction at the time of this change was
+    // 21.04 -> 14.78 using current ToT release build. (See also https://reviews.llvm.org/rL358304)
+    pto.ForgetAllSCEVInLoopUnroll = true;
+
+    // Tuning option, potentially target dependent. Set to halide needs after MemorySSA is turned on.
+    // pto.LicmMssaOptCap = 50;
```

Comment:
So these are intentionally commented out?

Reply:
Yes. If you prefer to remove entirely and add when setting those options, that's perfectly fine.

Reply:
It's fine, just add something to indicate more clearly they are deliberately commented out for now.

Reply:
I removed them for now to avoid the clutter. I'll send another PR after MSSA is turned on in LLVM.